# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/tibit/hexedit/00-hexedit-vi-keys.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

diff -u hexedit/interact.c hexedit-hacked/interact.c
--- hexedit/interact.c	Wed Jun 19 13:55:40 2002
+++ hexedit-hacked/interact.c	Wed Oct  9 12:04:18 2002
@@ -360,6 +360,39 @@
       next_line();
       break;
 
+      /* VI keys */
+    case 'h': 
+      if (!hexOrAscii) {
+	      setTo(key);
+	      break;
+      }
+      backward_char();
+      break;
+
+    case 'l': 
+      if (!hexOrAscii) {
+	      setTo(key);
+	      break;
+      }
+      forward_char();
+      break;
+
+    case 'k': 
+      if (!hexOrAscii) {
+	      setTo(key);
+	      break;
+      }
+      previous_line();
+      break;
+
+    case 'j': 
+      if (!hexOrAscii) {
+	      setTo(key);
+	      break;
+      }
+      next_line();
+      break;
+
     case KEY_UP:
     case CTRL('P'):
       previous_line();
@@ -443,6 +476,14 @@
 
     case CTRL('G'):
     case KEY_F(4):
+      goto_char();
+      break;
+
+    case 'g':
+      if (!hexOrAscii) {
+	      setTo(key);
+	      break;
+      }
       goto_char();
       break;
 
